hail/src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -156,6 +156,32 @@ object Region {
   def setMemory(offset: Code[Long], size: Code[Long], b: Code[Byte]): Code[Unit] =
     Code.invokeScalaObject[Long, Long, Byte, Unit](Region.getClass, "setMemory", offset, size, b)
 
+  def firstNonZeroByteOffset(address: Code[Long], nBits: Code[Long]): Code[Long] =
+    Code.invokeScalaObject[Long, Long, Long](Region.getClass, "firstNonZeroByteOffset", address, nBits)
+
+  def firstNonZeroByteOffset(address: Long, nBits: Long): Long = {
+    val m1 = nBits / 32L
+    var i = 0L
+
+    while(i < m1) {
+      if(loadInt(address + i * 4) != 0) {
```

Comment:
this isn't safe -- as written, your function doesn't assume any properties of `address`.
You could add a loop in front that goes by `byte`s to the first 8-byte-divisible address after `address`, then use longs.